test/dom-parsing/XMLSerializer.tests.ts

```diff
@@ -0,0 +1,346 @@
+import * as chai from 'chai';
+import * as slimdom from '../../src/index';
+
+import { appendAttribute } from '../../src/util/attrMutations';
+
+const HTML_NAMESPACE = 'http://www.w3.org/1999/xhtml';
+const XML_NAMESPACE = 'http://www.w3.org/XML/1998/namespace';
+const XMLNS_NAMESPACE = 'http://www.w3.org/2000/xmlns/';
+
+describe('XMLSerializer', () => {
+	let document: slimdom.Document;
+	let serializer: slimdom.XMLSerializer;
+	beforeEach(() => {
+		document = new slimdom.Document();
+		serializer = new slimdom.XMLSerializer();
+	});
+
+	it('returns the empty string if given an Attr', () => {
+		chai.assert.equal(serializer.serializeToString(document.createAttribute('test')), '');
+		chai.assert.equal(
+			serializer.serializeToString(document.createAttributeNS('http://www.example.com/ns', 'test')),
+			''
+		);
+	});
+
+	it('can serialize a CDATASection', () => {
+		chai.assert.equal(serializer.serializeToString(document.createCDATASection('test')), '<![CDATA[test]]>');
```

Comment:
CDataSections are special w.r.t serialization, though they are lightly used in our main dependant (FontoXML), it would be nice to add some tests for `createCDATASection(']]')`, `createCDATASection('&NOT_AN_ENTITY;')`, `createCDATASection('<element><!-- xml syntax in CData --></element>')`, etc?